File: Java/Procedural.java
package Paradigms.Java;

public class Procedural {
    public static class Point {
        public int x;
        public int y;
    }

    public static Point createPoint(int x, int y) {
        Point point = new Point();
        point.x = x;
        point.y = y;
        return point;
    }

    public static Point clone(Point point) {
        int x = point.x;
        int y = point.y;
        return createPoint(x, y);
    }

    public static void move(Point point, int x, int y) {
        point.x += x;
        point.y += y;
    }

    public static String toString(Point point) {
        int x = point.x;
        int y = point.y;
        return "(" + x + ", " + y + ")";
    }

    public static void main(String[] args) {
        Point p1 = createPoint(10, 20);
        System.out.println(toString(p1));
        Point c1 = clone(p1);
        move(c1, -5, 10);
        System.out.println(toString(c1));
    }
}

File: Python/procedural.py
def create_point(x, y):
    return {'x': x, 'y': y}

def clone(point):
    x = point['x']
    y = point['y']
    return create_point(x, y)

def move(point, dx, dy):
    point['x'] += dx
    point['y'] += dy

def to_string(point):
    x = point['x']
    y = point['y']
    return "({0}, {1})".format(x, y)

# Usage

p1 = create_point(10, 20)
print(to_string(p1))
c1 = clone(p1)
move(c1, -5, 10)
print(to_string(c1))
